### DISCUSSION TITLE
"code" block is missing support for shell script language

### DISCUSSION THREAD

**Discussion by anilv4** [standardnotes/forum#3452] [Suggestions]
**Have you read the [Longevity Statement](https://standardnotes.org/longevity)?**
Yes
**Is your feature request related to a problem? Please describe.**
I use code blocks to store many shell scripts that I use to manage my Linux servers.
As code blocks do not support shell language, the script looks very weird in code blocks.

**Describe the solution you'd like**
In code block, it should also show support for shell,sh, or bash, like javascript, c-like, etc

**Describe alternatives you've considered**
Right now, I am saving the shell scripts as plain text in the code block. It looks ugly, but at least avoid wrong highlights

**Additional context**

**Comment by GuilhermeGreggio** [comment]
Hey guys, I would really like to have this feature implemented.

I see that the app is using lexical/code for code blocks:
https://github.com/standardnotes/app/blob/main/packages/web/src/javascripts/Components/SuperEditor/Plugins/CodeOptionsPlugin/CodeOptions.tsx

And by default the lexical/code do include bash/sh support:
https://github.com/facebook/lexical/blob/main/packages/lexical-code-prism/src/FacadePrism.ts#L44

But I see that prism supports bash:
https://github.com/PrismJS/prism/blob/v2/src/languages/bash.js

So I was wondering if maybe we could import prism-bash in app/blob/main/packages/web/src/javascripts/Components/SuperEditor/Plugins/CodeOptionsPlugin/CodeOptions.tsx and expand the CODE_LANGUAGE_MAP, CODE_LANGUAGE_FRIENDLY_NAME_MAP.

I'm not a developer, so sorry if that doesn't make sense, just trying to help to get this feature in the app.
